### PR TITLE
Universal/deferred

### PR DESCRIPTION
**Pull request by kaychang-unity**
### Purpose of this PR
Deferred renderer support.

---
### Testing status

**Manual Tests**: What did you do?
- [ X ] Opened test project + Run graphic tests locally
- [ X ] Built a player

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/universal%252Fdeferred/.yamato%252Fupm-ci-abv.yml%2523all_project_ci_fast-trunk/1850211/job

Potential test projects* BoatAttack deferred branch, VikingVillage URP

---
### Comments to reviewers
This is revived from previous public SRP repository.

### CODE REVIEW

**Review by ernestasKupciunas — COMMENTED** [review]
Done with testing on mobiles.
3 bugs found in total:
[1235663 ](https://fogbugz.unity3d.com/f/cases/1235663/)[Deferred] [URP] [Gles3] Adreno 306 fails to render the scene (foggy view
[1234698](https://fogbugz.unity3d.com/f/cases/1234698/) [Deferred] Lights ignore Culling Mask setting
[1235660](https://fogbugz.unity3d.com/f/cases/1235660/) [Deferred] [URP] [Gles3] Distorted Real-time lights on Galaxy S6 Edge

Full testing Doc: https://docs.google.com/document/d/17SlB6XVU-70PzIa44SEx26YK-nWfSVQi-8y-p2OB64M/edit?usp=sharing
Keep in mind, that I was testing Deferred renderer together with Render Pass, and bugs that was specifically related to Deferred are in **Deferred renderer /trunk issues(Not related to Render Pass)** section.
Projects used for testing:
[Winter](https://github.com/ernestasKupciunas/Winter) 
[Vikings Village URP](https://github.com/Jonasmortensen/VikingVillageURP)
[Boat Attack Deferred](https://github.com/Verasl/BoatAttack/tree/deferred-renderer)

Seems like only Android side had some issues. iOS device rendered all testing projects without problems using Deferred renderer

**Review by ernestasKupciunas — APPROVED** [review]
Besides some issues mentioned above, I think Deferred renderer works pretty stable on mobile devices

**Comment by kaychang-unity** [comment]
@phi-lira @ellioman I merged with master again.

**Comment by kaychang-unity** [comment]
@phi-lira should I take out reviewers who are not responding (@Unity-Technologies/shader-graph , @Unity-Technologies/2d-graphics, @Unity-Technologies/gfx-qa-bellevue ) ?

**Review by alindmanUnity — APPROVED** [review]

**Review by unity-cchu — APPROVED** [review]

File: com.unity.render-pipelines.universal/Runtime/2D/Passes/Render2DLightingPass.cs [hunk 1]
```diff
@@ -11,9 +11,10 @@ internal class Render2DLightingPass : ScriptableRenderPass
         Renderer2DData m_Renderer2DData;
         static readonly ShaderTagId k_CombinedRenderingPassNameOld = new ShaderTagId("Lightweight2D");
         static readonly ShaderTagId k_CombinedRenderingPassName = new ShaderTagId("Universal2D");
+        static readonly ShaderTagId k_UnlitPassName = new ShaderTagId("Universal2DUnlit");
```
Comment (rainsing) [inline com.unity.render-pipelines.universal/Runtime/2D/Passes/Render2DLightingPass.cs:14]: Is this new light mode tag ever used in any of the new or updated shaders? A solution scope search didn't show anything.

**Review by rainsing — APPROVED** [review]

**Review by phi-lira — APPROVED** [review]

**Pull request merged**